### PR TITLE
FSPT-759 Single level of nested groups

### PR DESCRIPTION
**Pull request by srh-sloan**
## 🎫 Ticket
[FSPT-759 Allow single level of nested groups](https://mhclgdigital.atlassian.net/browse/FSPT-759)

## 📝 Description
Allows form designers to add a single level of nested question groups. For example, inside a conditional group of questions, you can have one or more groups, but no groups further down than that.

Notes on restrictions:
- We have decided to implement a restriction on the number of levels of nesting to only allow one level of sub groups. This is done via a configuration value `MAX_NESTED_GROUP_LEVELS` so can easily be updated, but tests added to remind devs to check everything still works if they update this number.
- The UI will not let a form designer add a question group if it would create too many levels of nested groups (wording of the text for this still TBC with design)
- They also can't create a group if the parent group is set to show all questions on the same page.
- The above trigger custom exceptions, but we don't surface these to the user as they should never hit these situations.

## 📸 Show the thing (screenshots, gifs)
https://github.com/user-attachments/assets/2e4fcae7-4bd8-4cf6-876d-ecffe0d0efd1

## 🧪 Testing
- Unit tests added
- Tested locally both using the UI and by showing buttons that shouldn't be there to confirm error conditions are met

### Review Readiness
- [ x] PR title and description provide sufficient context for reviewers
- [ x] Commits are logical, self-contained, and have clear descriptions

### Performance and security
- [x ] No N+1 query problems introduced
- [ ] Any new DB queries include appropriate where clauses based on the user's permissions

### Testing
- [ x] I have tested this change and it meets the acceptance criteria for the ticket
- [ x] I need the reviewer(s) to pull and run this change locally
- [ x] New (non-developer) functionality has appropriate unit and integration tests
- [ ] End-to-end tests have been updated (if applicable)
- [ x] Edge cases and error conditions are tested

## 📚 Additional Notes
Still to do:
- Breadcrumb (use ellipsis to show multiple interim levels)
- e2e tests
- seed grants with nested groups

### CODE REVIEW

File: app/common/data/interfaces/collections.py [hunk 1]
```diff
@@ -402,6 +403,21 @@ def create_group(
...
+                "You cannot create a nested group at this level",
+                parent_group=parent,
+                child_group_name=name or text,
+                nesting_level=parent.nested_group_levels + 1,
+            )
+        if parent.same_page:
+            raise NestedGroupDisplayTypeSamePageException(
+                "You cannot create a nested group if the parent is set to display all questions on the same page",
+                parent_group=parent,
+            )
```
[the remark this answers is not in the document]
Comment (samuelhwilliams) [reply app/common/data/interfaces/collections.py:420]: Also, it's possible to _view_ the 'create group' flow pages even if the parent group has the same-page setting. Maybe we could short-circuit some errors being raised by checking for that condition and 400'ing the pages directly (or just redirecting back to the parent group if we don't want to show an error page).

File: app/common/data/interfaces/collections.py [hunk 2]
```diff
@@ -402,6 +403,21 @@ def create_group(
     parent: Optional[Group] = None,
     presentation_options: QuestionPresentationOptions | None = None,
 ) -> Group:
+    # If this group is nested, ensure it meets rules for nesting groups
+    # This is a safety check as we don't allow users to create nested groups when these rules aren't met
+    if parent:
+        if not parent.is_allowed_nested_groups:
```
Comment (samuelhwilliams) [inline app/common/data/interfaces/collections.py:399]: optional nit: The naming of this just caused me a 🤏  bit of confusion - it sounds like that attribute is answering whether or not you can create a nested group at all (which we have two logical rules for: # of levels of nesting, and the parent being same page). This attribute only checks the level of nesting.

**Review by samuelhwilliams — APPROVED** [review]
Nice - love the error handling, feels really robust. Good test coverage of the interfaces as well :)

**Review by samuelhwilliams — APPROVED** [review]
